app/assets/javascripts/foundation_extras.js

```diff
@@ -3,6 +3,13 @@
   App.FoundationExtras = {
     initialize: function() {
       $(document).foundation();
+    },
+    destroy: function() {
+      if ($(".sticky").length > 0) {
+        $(".sticky").foundation("_destroy");
+      }
     }
   };
+
+  $(document).on("turbolinks:before-visit", App.FoundationExtras.destroy);
```

Comment:
It works exactly the same if i add `App.FoundationExtras.destroy();` to  `destroy_non_idempotent_modules` function as you suggest but i did not understood this error as an idempotency problem, this error is because of the combination we are using, Turbolinks + Foundation, Turbolinks replaces the document body leaving the window object as it was at previous page so scrollListener added by Foundation keeps being executed at the new page where Sticky is not needed. So the reason to put this here was only my understanding of the problem. What do you think?

Reply:
I see :thinking:. Like this it's fine with me :smile:.